### PR TITLE
[Search] Improve TestPipeline instantiation options

### PR DESCRIPTION
**Pull request by iamEAP**
## What/Why

It turns out that the magic behind `TestPipeline.withSubject()` is no match for sufficiently complex test subjects.  For instance, if one makes a collator that is actually a `PassThrough` stream that combines multiple input streams...  It's impossible to test, because `withSubject()` thinks it's a decorator.

The path forward seems to be to deprecate the magic method in favor of more explicit equivalents.  ...I'm also taking the opportunity to differentiate between `from*` being for static instantiation and `with*` being for fluent instance modification.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/search-backend-node/src/test-utils/TestPipeline.ts [hunk 1]
```diff
@@ -79,6 +111,48 @@ export class TestPipeline {
...
+   * Create a test pipeline given a collator you want to test.
+   */
+  static fromCollator(collator: Readable) {
+    return new TestPipeline({ collator });
+  }
+
+  /**
+   * Add a collator to the test pipeline.
+   */
+  withCollator(collator: Readable) {
```
Comment (freben) [inline plugins/search-backend-node/src/test-utils/TestPipeline.ts:124]:
```suggestion
  withCollator(collator: Readable): this {
```

and then returning `this` of course, in all of these `withX` methods.

**Review by freben — APPROVED** [review]
Looks good, just one QoL idea

File: .changeset/search-tofu-panaeng.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-search-backend-node': patch
+---
+
+Use of `TestPipeline.withSubject()` is now deprecated. Instead, use the `fromCollator`, `fromDecorator`, or `fromIndexer` static methods to instantiate a test pipeline. You may also use the class' `withCollator`, `withDecorator`, and `withIndexer` instance methods to build test pipelines that consist of multiple test subjects.
```
Comment (emmaindal) [inline .changeset/search-tofu-panaeng.md:5]: maybe good to add something around expected time of deletion? E.g.

```
Use of `TestPipeline.withSubject()` is now deprecated and will be deleted in next main release
```

**Review by emmaindal — APPROVED** [review]
just one suggestion for the changeset, otherwise looks good!

File: .changeset/search-tofu-panaeng.md (hunk 2 above)
Comment (iamEAP) [reply .changeset/search-tofu-panaeng.md:5]: I'm feeling a little non-committal on a timeline for actually removing it; I figure sometime in the (fairly distant) future, we can do one big sweep of `@deprecated` code, remove all instances, and do a single `major` bump to `2.0.0` that covers all of them.  As it is, I think this is the only one, so far.

Comment (emmaindal) [reply .changeset/search-tofu-panaeng.md:5]: Make sense 👍

**Pull request merged**